mmdet/datasets/pipelines/loading.py

```diff
@@ -76,6 +76,125 @@ def __repr__(self):
         return repr_str
 
 
+@PIPELINES.register_module()
+class LoadMultiImagesFromMultiFiles(object):
+    """Load multiple images from multiple files.
+
+    Required keys are "img_prefix" and img_info_keys (dicts that must
+    contain the key "filename").
+    (`img_info_keys = [prefix + '_imgs_info' for prefix in prefixs]`)
+    Added or updated keys are "*filenames",
+    "*ori_filenames", "*img*", "img_shape", "ori_shape" (same as `img_shape`),
+    "pad_shape" (same as `img_shape`), "scale_factor" (1.0) and "img_norm_cfg"
+    (means=0 and stds=1).
+
+    Args:
+        prefixs (list): the prefix keys of img info.
+            Defaults to ['target', 'ref'].
+        to_float32 (bool): Whether to convert the loaded image to a float32
+            numpy array. If set to False, the loaded image is an uint8 array.
+            Defaults to False.
+        color_type (str): The flag argument for :func:`mmcv.imfrombytes()`.
+            Defaults to 'color'.
+        file_client_args (dict): Arguments to instantiate a FileClient.
+            See :class:`mmcv.fileio.FileClient` for details.
+            Defaults to ``dict(backend='disk')``.
+    """
+
+    def __init__(self,
+                 prefixs=['target', 'ref'],
+                 to_float32=False,
+                 color_type='color',
+                 file_client_args=dict(backend='disk')):
+        self.prefixs = prefixs.copy()
+        self.to_float32 = to_float32
+        self.color_type = color_type
+        self.file_client_args = file_client_args.copy()
+        self.file_client = None
+
+    def load_img_core(self, results, fname_id, fname, ori_fname, img_prefix):
+        img_bytes = self.file_client.get(fname)
+        img = mmcv.imfrombytes(img_bytes, flag=self.color_type)
+        if self.to_float32:
+            img = img.astype(np.float32)
+
+        # e.g. img_prefix = 'target' or 'ref'
+        if f'{img_prefix}_filenames' not in results:
+            results[f'{img_prefix}_filenames'] = [fname]
+        else:
+            results[f'{img_prefix}_filenames'].append(fname)
```

Comment:
Only use `results[f'{img_prefix}_filenames'].append(fname)` and declare `results[f'{img_prefix}_filenames']=[]` in __call__. Do the same for `results[f'{img_prefix}_ori_filenames']`

Reply:
The key of f'{img_prefix}_filenames' (e.g. 'target_filenames') may not exist in VID testing, therefore I prefer to use 'if else' rather than init an empty list for results['target_filenames']

Reply:
Done